### ISSUE TITLE
Parse error when constructor argument has negative number default value

### ISSUE THREAD

**Issue by dishuostec** [alangpierce/sucrase#522]
This code will produce an error `Expected identifier after access modifiers in constructor arg.`:
```typescript
export class Foo {
	constructor({ p = -1 }: { p?: number }) {

	}
}
```

see https://sucrase.io/#selectedTransforms=typescript&code=export%20class%20Foo%20%7B%0A%09constructor(%7B%20p%20%3D%20-1%20%7D%3A%20%7B%20p%3A%20number%20%7D)%20%7B%0A%0A%09%7D%0A%7D

**Comment by alangpierce** [comment]
Thanks for the report! Looks like it's misinterpreting the `-` as a flow access modifier. I've got a fix working and I should be able to release it soon.

**Comment by alangpierce** [comment]
This is fixed now in Sucrase 3.14.0.
